sda/websocket.go

```diff
@@ -0,0 +1,366 @@
+package sda
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"time"
+
+	"reflect"
+	"strings"
+
+	"sync"
+
+	"github.com/dedis/cothority/log"
+	"github.com/dedis/cothority/network"
+	"github.com/dedis/protobuf"
+	"github.com/gorilla/websocket"
+	"gopkg.in/tylerb/graceful.v1"
+)
+
+// WebSocket handles incoming client-requests using the WebSocket
+// protocol for JavaScript-compatibility. When making a new WebSocket,
```

Comment:
We are implementing a websocket server because we need a external way for client to communicate with a cothority node. It just so happens that websocket is the most inter-operable way to communicate with language such as javascript but I would not put it as the main reason. 
Sorry, nitpick as always.